### PR TITLE
fix: Continue syncing if no users are found

### PR DESCRIPTION
**Pull request by martinalbert**
This PR implements more robust flow of syncing users and roles. If no user ids are found, we continue syncing other resources.

This PR also updates dependencies and downgrades go-falcon library due to incorrect mapping of domain_user. (issue described here: https://github.com/CrowdStrike/gofalcon/issues/425

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,113 +1,115 @@
 module github.com/conductorone/baton-crowdstrike
 
-go 1.20
+go 1.22
 
 require (
-	github.com/conductorone/baton-sdk v0.1.11
-	github.com/crowdstrike/gofalcon v0.4.2
+	github.com/conductorone/baton-sdk v0.1.33
+	github.com/crowdstrike/gofalcon v0.3.1
```
Comment (ggreer) [inline go.mod:7]: It looks like a release candidate has the fix: https://github.com/CrowdStrike/gofalcon/releases/tag/0.7.0-rc1

Do we want to try using that? If it works, we can switch to 0.7.0 when it comes out.

File: pkg/connector/helpers.go [hunk 2]
```diff
@@ -75,8 +75,8 @@ type RateLimitInfo struct {
 	Remaining int64
 }
 
-func NewRateLimitInfo(limit, remaining int64) RateLimitInfo {
-	return RateLimitInfo{
+func NewRateLimitInfo(limit, remaining int64) *RateLimitInfo {
+	return &RateLimitInfo{
```
Comment (ggreer) [inline pkg/connector/helpers.go:79]: Why is this change needed? The code looks uglier so it must be important. :)

File: go.mod (hunk 1 above)
Comment (martinalbert) [reply go.mod:7]: Seems like the github release went through, but I can't find it in the go registry: https://pkg.go.dev/github.com/crowdstrike/gofalcon?tab=versions. Will let them know.

File: pkg/connector/helpers.go (hunk 2 above)
Comment (martinalbert) [reply pkg/connector/helpers.go]: @ggreer it is not needed, I just like to use pointers as a good practice 😄 I'm not sure about using it with such small structs though. I reverted this change and use copies of `RateLimitInfo` struct. I could refactor it to use array of pointers, but that might be overkill, what do you think of using pointers over copies?

File: go.mod (hunk 1 above)
Comment (martinalbert) [reply go.mod]: Was not aware of installation steps of pre-releases, fixed in 84dacf5406f9ef014b1bd5a1662c06b267e221e9
